bigquery_etl/glean_usage/common.py

```diff
@@ -113,88 +141,113 @@ def _extract_dataset_from_glob(pattern):
-    """Generate and run queries using a threadpool.
+    return app_info
 
-    The callback function is reponsible for generating and running the queries.
-    This was the original main entrypoint for each of the usage queries.
-    """
-    parser = get_argument_parser(description)
-    args = parser.parse_args()
-
-    try:
-        logging.basicConfig(level=args.log_level, format="%(levelname)s %(message)s")
-    except ValueError as e:
-        parser.error(f"argument --log-level: {e}")
-
-    baseline_tables = list_baseline_tables(
-        project_id=args.project_id,
-        only_tables=getattr(args, "only_tables", None),
-        table_filter=args.table_filter,
-    )
 
-    with ThreadPool(args.parallelism) as pool:
-        # Do a first pass with dry_run=True so we don't end up with a partial success;
-        # we also write out queries in this pass if so configured.
-        pool.map(
-            partial(
-                run_query_callback,
-                args.project_id,
-                date=args.date,
-                dry_run=True,
-                output_dir=args.output_dir,
-                output_only=args.output_only,
-            ),
-            baseline_tables,
+class GleanTable:
+    """Represents a generated Glean table."""
+
+    def __init__(self):
+        """Init Glean table."""
+        self.target_table_id = ""
+        self.prefix = ""
+        self.custom_render_kwargs = {}
+        self.no_init = True
+
+    def generate_per_app_id(self, project_id, baseline_table, output_dir=None):
+        """Generate the baseline table query per app_id."""
+        tables = table_names_from_baseline(baseline_table, include_project_id=False)
+
+        init_filename = f"{self.target_table_id}.init.sql"
+        query_filename = f"{self.target_table_id}.query.sql"
+        view_filename = f"{self.target_table_id[:-3]}.view.sql"
+        view_metadata_filename = f"{self.target_table_id[:-3]}.metadata.yaml"
+
+        table = tables[f"{self.prefix}_table"]
+        view = tables[f"{self.prefix}_view"]
+        render_kwargs = dict(
+            header="-- Generated via bigquery_etl.glean_usage\n",
+            project_id=project_id,
         )
-        if args.output_only:
+
+        render_kwargs.update(self.custom_render_kwargs)
+        render_kwargs.update(tables)
+
+        query_sql = render(query_filename, **render_kwargs)
+        view_sql = render(view_filename, **render_kwargs)
+        view_metadata = render(view_metadata_filename, format=False, **render_kwargs)
+
+        if not self.no_init:
+            try:
+                init_sql = render(init_filename, **render_kwargs)
+            except TemplateNotFound:
+                init_sql = render(query_filename, init=True, **render_kwargs)
+
+        if not (referenced_table_exists(view_sql)):
+            logging.info(
+                "Skipping view for table which doesn't exist:"
+                f" {self.target_table_id}"
```

Comment:
This should include a qualifier that indicates which app_id/dataset is involved.